wire-ios-data-model/Source/UseCases/HasSelfUserValidE2EICertificatesForAllClientsUseCase.swift

```diff
@@ -0,0 +1,92 @@
+//
+// Wire
+// Copyright (C) 2024 Wire Swiss GmbH
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with this program. If not, see http://www.gnu.org/licenses/.
+//
+
+import Foundation
+import WireCoreCrypto
+
+// sourcery: AutoMockable
+/// Determines if the self user has a valid E2EI certificate on all clients.
+public protocol HasSelfUserValidE2EICertificatesForAllClientsUseCaseProtocol {
+    /// Returns `true` if all clients of the self user have valid E2EI certificates.
+    func invoke() async throws -> Bool
+}
+
+public struct HasSelfUserValidE2EICertificatesForAllClientsUseCase: HasSelfUserValidE2EICertificatesForAllClientsUseCaseProtocol {
+
+    private let context: NSManagedObjectContext
+    private let coreCryptoProvider: CoreCryptoProviderProtocol
+
+    public init(
+        context: NSManagedObjectContext,
+        coreCryptoProvider: CoreCryptoProviderProtocol
+    ) {
+        self.context = context
+        self.coreCryptoProvider = coreCryptoProvider
+    }
+
+    public func invoke() async throws -> Bool {
+        let (conversationID, userID) = try await context.perform {
+            // conversationID
+            guard let selfConversation = ZMConversation.fetchSelfMLSConversation(in: context) else {
+                throw Error.couldNotFetchMLSSelfConversation
+            }
+            guard let mlsGroupID = selfConversation.mlsGroupID else {
+                throw Error.failedToGetMLSGroupID(selfConversation)
+            }
+            // userID
+            let selfUser = ZMUser.selfUser(in: context)
+            guard
+                let qualifiedID = selfUser.qualifiedID,
+                let userID = Optional(qualifiedID.uuid.transportString()) // TODO: [WPB-765]: workaround for a bug in core crypto, should be fixed mid february 2024, no JIRA ticket
```

Comment:
I think we don't need to make in `Optional`, right?

Reply:
Since I want to replace this line with the commented line below once Core Crypto is upgraded to a fixed version, I would like to keep the line in the guard-statement.
For that it must be optional.
Do you mind keeping it for a couple of days?

Reply:
Ok, let's keep it